database/DevHome.Database/Services/RepositoryManagementDataAccessService.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using DevHome.Common.TelemetryEvents.DevHomeDatabase;
+using DevHome.Database.DatabaseModels.RepositoryManagement;
+using DevHome.Database.Factories;
+using DevHome.Telemetry;
+using Serilog;
+
+namespace DevHome.Database.Services;
+
+public class RepositoryManagementDataAccessService
+{
+    private const string EventName = "DevHome_RepositoryData_Event";
+
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(RepositoryManagementDataAccessService));
+
+    private readonly DevHomeDatabaseContextFactory _databaseContextFactory;
+
+    public RepositoryManagementDataAccessService(DevHomeDatabaseContextFactory databaseContextFactory)
+    {
+        _databaseContextFactory = databaseContextFactory;
+    }
+
+    /// <summary>
+    /// Makes a new Repository entity with the provided name and location then saves it
+    /// to the database.
+    /// </summary>
+    /// <param name="repositoryName">The name of the repository to add.</param>
+    /// <param name="cloneLocation">The local location the repository is cloned to.</param>
+    /// <returns>The new repository.  Can return null if the database threw an exception.</returns>
+    public Repository? MakeRepository(string repositoryName, string cloneLocation, Uri repositoryUri)
+    {
+        return MakeRepository(repositoryName, cloneLocation, string.Empty, repositoryUri);
+    }
+
+    public Repository? MakeRepository(string repositoryName, string cloneLocation, string configurationFileLocationAndName, Uri repositoryUri)
+    {
+        var existingRepository = GetRepository(repositoryName, cloneLocation);
+        if (existingRepository != null)
+        {
+            _log.Information($"A Repository with name {repositoryName} and clone location {cloneLocation} exists in the repository already.");
```

Comment:
In a since, I understand how this is a failure.  If the repository did exist then no repository was made.

I will change this to a warning.  I do not want to make this an error because a user can't "fix" the error.